module/Company/src/Company/Service/Company.php

```diff
@@ -56,12 +56,25 @@ public function getHiddenCompanyList()
      * @param mixed $package
      * @param mixed $data
      */
-    public function savePackageByData($package,$data)
+    public function savePackageByData($package, $data, $files)
     {
         $packageForm = $this->getPackageForm();
         $packageForm->setData($data);
         if ($packageForm->isValid()){
-            $package->exchangeArray($data);
+            if ($package->getType() == 'banner'){
+                $package->exchangeArray($data);
+                $file = $files['banner'];
+                try {
+                    $oldPath = $package->getImage();
+                    $newPath = $this->getFileStorageService()->storeUploadedFile($file);
+                    $package->setImage($newPath);
+                    if ($oldPath != '' && oldPath != newPath) {
+                        $this->getFileStorageService()->removeFile($oldPath);
+                    }
+                } catch (\Exception $exception) {
```

Comment:
I need to catch an \Exception here. Sure, I could rewrite this part to check for the error condition too, but that would result in code duplication, which is also bad practise.